### PR TITLE
whisper-cpp 1.7.3

### PR DESCRIPTION
**Pull request by mislav**
https://github.com/ggerganov/whisper.cpp/releases/tag/v1.7.3

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/w/whisper-cpp.rb [hunk 1]
```diff
@@ -22,11 +22,27 @@ class WhisperCpp < Formula
...
+      -DGGML_METAL=#{(OS.mac? && !Hardware::CPU.intel?) ? "ON" : "OFF"}
+      -DGGML_METAL_EMBED_LIBRARY=#{OS.mac? ? "ON" : "OFF"}
+      -DGGML_NATIVE=#{build.bottle? ? "OFF" : "ON"}
+      -DWHISPER_BUILD_EXAMPLES=ON
+      -DWHISPER_BUILD_TESTS=OFF
+      -DWHISPER_BUILD_SERVER=OFF
+    ]
+    args << "-DLLAMA_METAL_MACOSX_VERSION_MIN=#{MacOS.version}" if OS.mac?
+
+    system "cmake", "-B", "build", *args, *std_cmake_args
```
Comment (carlocab) [inline Formula/w/whisper-cpp.rb:39]: Let's follow the style used by other formulae:
```suggestion
    system "cmake", "-S", ".", "-B", "build", *args, *std_cmake_args
```

File: Formula/w/whisper-cpp.rb [hunk 2]
```diff
@@ -22,11 +22,27 @@ class WhisperCpp < Formula
...
+    # the "main" target is our whisper-cpp binary
+    system "cmake", "--build", "build", "--target", "main"
+    bin.install "build/bin/main" => "whisper-cpp"
```
Comment (carlocab) [inline Formula/w/whisper-cpp.rb:42]: Is there no `install` target yet? `llama.cpp` seems to have one already these days.

File: Formula/w/whisper-cpp.rb [hunk 3]
```diff
@@ -22,11 +22,27 @@ class WhisperCpp < Formula
...
+
+    mkdir prefix/"test"
+    mv ["models/for-tests-ggml-tiny.bin", "samples/jfk.wav"], prefix/"test"
```
Comment (carlocab) [inline Formula/w/whisper-cpp.rb:45]: We usually install files used for testing into `pkgshare`:
```suggestion
    pkgshare.install "models/for-tests-ggml-tiny.bin", "samples/jfk.wav"
```

**Review by carlocab — COMMENTED** [review]
Seems ok to me; a few comments.

File: Formula/w/whisper-cpp.rb (hunk 2 above)
Comment (mislav) [reply Formula/w/whisper-cpp.rb:42]: There is an install target but at the moment it only installs dylibs (had `BUILD_SHARED_LIBS` been left on) but not the `main` binary. I've submitted this as feedback to the devs and they are already updating their build system, so I expect this to improve in future releases.

**Comment by mislav** [comment]
@carlocab Thanks; I've applied the suggestions. BTW, I'm force-pushing each time because I got feedback from homebrew-core maintainers in the past that you prefer PRs having a single commit and that your tooling does not allow squashed merges. Hope that's okay.

**Review by p-linnane — APPROVED** [review]

File: Formula/w/whisper-cpp.rb (hunk 2 above)
Comment (daeho-ro) [reply Formula/w/whisper-cpp.rb:42]: Then, we can just ship it as-is first and update it later.

File: Formula/w/whisper-cpp.rb [hunk 4]
```diff
@@ -22,11 +22,27 @@ class WhisperCpp < Formula
...
+      -DWHISPER_BUILD_SERVER=OFF
+    ]
+    args << "-DLLAMA_METAL_MACOSX_VERSION_MIN=#{MacOS.version}" if OS.mac?
+
+    system "cmake", "-S", ".", "-B", "build", *args, *std_cmake_args
+    # the "main" target is our whisper-cpp binary
+    system "cmake", "--build", "build", "--target", "main"
+    bin.install "build/bin/main" => "whisper-cpp"
+
+    mkdir prefix/"test"
```
Comment (carlocab) [inline Formula/w/whisper-cpp.rb:44]:
```suggestion
```

**Review by carlocab — APPROVED** [review]

**Review by carlocab — APPROVED** [review]
Thanks @mislav! Great work as always.

**Review by carlocab — APPROVED** [review]

**Pull request merged**